### PR TITLE
Add URI#authority

### PR DESCRIPTION
**Pull request by straight-shoota**
Adds a method for querying the authority component of a URI.

Ref: https://github.com/crystal-lang/crystal/pull/9990#issuecomment-735270515

### CODE REVIEW

File: src/uri.cr [hunk 1]
```diff
@@ -250,26 +250,53 @@ class URI
...
-  def to_s(io : IO) : Nil
-    if scheme
-      io << scheme
-      io << ':'
+  # Returns the authority component of this URI.
+  # It is formatted as `user:pass@host:port` with missing parts being omitted.
+  #
+  # If the URI does not have any authority information, the result is `nil`.
+  #
+  # ```cr
```
Comment (Sija) [inline src/uri.cr:258]:
```suggestion
  # ```
```

**Review by jhass — APPROVED** [review]

File: src/uri.cr (hunk 1 above)
Comment (Sija) [reply src/uri.cr:258]: @straight-shoota Please, remove this before merging.

Comment (asterite) [reply src/uri.cr:258]: If someone wants to do it, we can have the formatter remove these "cr" things.

Comment (straight-shoota) [reply src/uri.cr:258]: @asterite Already done, going to make a PR soon

**Review by asterite — APPROVED** [review]

**Pull request merged**